File: dotnet/DesignPatterns/BehavioralDesignPatterns/ChainOfResponsibilityDesignPattern/ChainOfResponsibilityDesignPattern.Example3/Handlers/Level1SupportHandler.cs
namespace ChainOfResponsibilityDesignPattern.Example3.Handlers;

public class Level1SupportHandler : BaseSupportHandler
{
    public Level1SupportHandler() : base("Level 1 Support") { }

    public override void HandleTicket(SupportTicket ticket)
    {
        // Level 1 can handle Low priority tickets and basic General issues
        if (CanHandle(ticket, TicketPriority.Low, new List<TicketCategory> { TicketCategory.General }))
        {
            ticket.Resolve(HandlerName, "Provided basic troubleshooting steps and FAQ links.");
        }
        else
        {
            base.HandleTicket(ticket);
        }
    }
}

File: nodejs/DesignPatterns/BehavioralDesignPatterns/ChainOfResponsibilityDesignPattern/ChainOfResponsibilityDesignPattern.Example3/src/Handlers/Level1SupportHandler.ts
import { SupportTicket, TicketCategory, TicketPriority } from "../SupportTicket";
import { BaseSupportHandler } from "../BaseSupportHandler";

export class Level1SupportHandler extends BaseSupportHandler {
  constructor() { super("Level 1 Support"); }

  handleTicket(ticket: SupportTicket): void {
    if (this.canHandle(ticket, TicketPriority.Low, [TicketCategory.General])) {
      ticket.resolve(this.handlerName, "Provided basic troubleshooting steps and FAQ links.");
    } else {
      super.handleTicket(ticket);
    }
  }
}
